service-base/src/test/java/org/eclipse/hono/service/management/device/AutoProvisioningEnabledDeviceBackendTest.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2020 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+
+package org.eclipse.hono.service.management.device;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.doAnswer;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.cert.CertificateFactory;
+import java.security.cert.X509Certificate;
+import java.util.Optional;
+
+import org.eclipse.hono.service.management.Id;
+import org.eclipse.hono.service.management.OperationResult;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+
+import io.opentracing.noop.NoopSpan;
+import io.vertx.core.Future;
+import io.vertx.core.Promise;
+import io.vertx.core.net.SelfSignedCertificate;
+import io.vertx.junit5.VertxExtension;
+import io.vertx.junit5.VertxTestContext;
+
+/**
+ * Tests verifying behavior of {@link AutoProvisioningEnabledDeviceBackend}.
+ */
+@ExtendWith(VertxExtension.class)
+class AutoProvisioningEnabledDeviceBackendTest {
```

Comment:
can you please make this public as all our other test classes?

Reply:
Sure. I was not aware of this convention. Since public classes and methods are no longer necessary with JUnit 5 and I found a couple of tests in Hono that are not public, I assumed we would move in that direction.

Reply:
Do we? I was not aware of this. In any case, the checkstyle rules for JavaDoc do not kick in for package private classes and methods but IMHO we should enforce proper JavaDocs for test cases as well ...

Reply:
Maybe we could change the checkstyle rules to enforce JavaDocs for package private classes and methods in the tests like this https://stackoverflow.com/a/25898007/7740194 
I like the possibility to have less boilerplate and visual clutter and there is no real need for a higher visibility level. WDYT?

Reply:
I think that this is a lot of effort for you to be able to save typing *public* ;-)
When I create a test case, the (public) class is automatically generated by Eclipse IDE and I only need to add the test methods ...

Reply:
> When I create a test case, the (public) class is automatically generated by Eclipse IDE and I only need to add the test methods ...

I'm sorry for you, when I create a (JUnit 5) test case, the package private class is automatically generated by the IntelliJ IDE along with the package private test methods and I just have to add the logic ;-) 
Saving typing is not really worth it. Having a cleaner code might be worth it.

Reply:
I do not really see why the code would become *cleaner* if all classes and/or methods are package private instead of public. But maybe the real problem is that you cannot configure IntelliJ to create them as public in the first place? ;-)